### PR TITLE
Filter only published posts

### PR DESCRIPTION
**Pull request by hvanoch**
I am targeting this branch, because of bugfix.
## Changelog
```markdown
### Fixed
- filter post by publishing time to prevent listing unpublished posts
```
## Subject

Unpublished post are listed in the pager. When clicking on the link it shows a 404.

### CODE REVIEW

File: Entity/PostManager.php [hunk 1]
```diff
@@ -96,7 +96,10 @@ public function getPager(array $criteria, $page, $limit = 10, array $sort = arra
         $query = $this->getRepository()
             ->createQueryBuilder('p')
             ->select('p, t')
+            ->andWhere('p.publicationDateStart <= :now')
             ->orderBy('p.publicationDateStart', 'DESC');
+        
+        $parameters['now'] = new \DateTime();
```
Comment (greg0ire) [inline Entity/PostManager.php]: Why not use `NOW()` directly? Not portable enough?

Comment (core23) [inline Entity/PostManager.php]: This won't work with Oracle

Comment (greg0ire) [inline Entity/PostManager.php]: Isn't doctrine able to map to a different method? See https://github.com/doctrine/dbal/search?utf8=%E2%9C%93&q=getNowExpression&type=

Comment (hvanoch) [inline Entity/PostManager.php]: I also don't like using the "mysql" time. 
What if database and application are on different servers, with different timezones. (not common, but possible)

Comment (greg0ire) [inline Entity/PostManager.php]: Good point. Maybe stick with that indeed.

**Comment by greg0ire** [comment]
@hvanoch do you need help fixing the build?

**Comment by hvanoch** [comment]
@greg0ire I just applied the patch from Scrutinizer.

Is there anything else I need to do?

**Comment by greg0ire** [comment]
@hvanoch yes, you need to fix [The Travis build](https://travis-ci.org/sonata-project/SonataNewsBundle/builds/243151296?utm_source=github_status&utm_medium=notification), it is required. Do you need help doing that?

**Comment by OskarStark** [comment]
@hvanoch please finish this PR, it should be easy

File: Entity/PostManager.php [hunk 2]
```diff
@@ -96,8 +96,11 @@ public function getPager(array $criteria, $page, $limit = 10, array $sort = arra
         $query = $this->getRepository()
             ->createQueryBuilder('p')
             ->select('p, t')
+            ->andWhere('p.publicationDateStart <= :now')
             ->orderBy('p.publicationDateStart', 'DESC');
 
+        $parameters['now'] = new \DateTime();
```
Comment (core23) [inline Entity/PostManager.php]: Maybe use the criteria for that with a default value?

Comment (core23) [reply Entity/PostManager.php]: Ping @hvanoch

Comment (core23) [reply Entity/PostManager.php]: Ping @hvanoch

**Comment by OskarStark** [comment]
Closing this due to inactivity

**Pull request closed without merging**